### ISSUE TITLE
What is the meaning of the value returned by /debug endpoint?

### ISSUE THREAD

**Issue by thevops** [cculianu/Fulcrum#163]
Hi,
I didn't find the answer so I'm asking here.
What means the below value? Is it a performance metric? For sure, it's a time, but I don't know what that time describes.

```
❯ curl -s localhost:8080/debug
{
    "elapsed": "0.007215 msec"
}
```

**Comment by thevops** [comment]
Thanks a lot! Helpful information.

I've tried to do it, but I got some issues:
```
debuger:~# curl localhost:8080/debug
{
    "elapsed": "0.007380 msec"
}
debuger:~# curl localhost:8080/debug?mempool=1
{
    "error": "Unable to obtain a result within the time period specified"
}
debuger:~# curl localhost:8080/debug
{
    "elapsed": "0.005743 msec"
}
debuger:~# curl localhost:8080/debug?utxo_stats=1
{
    "error": "Unable to obtain a result within the time period specified"
}
debuger:~# curl localhost:8080/debug?subs=1
{
    "error": "Unable to obtain a result within the time period specified"
}
debuger:~# curl localhost:8080/debug?dspsubs=1
{
    "error": "Unable to obtain a result within the time period specified"
}
debuger:~# curl localhost:8080/debug?txsubs=1
{
    "error": "Unable to obtain a result within the time period specified"
}
debuger:~# curl localhost:8080/debug
{
    "error": "Unable to obtain a result within the time period specified"
}
```

It has not once succeeded. Just simple `/debug` works, but not every time.
It looks like there is too little timeout. Is there any option to increase it?

**Comment by cculianu** [comment]
Yeah well on a server with clients connected that is busy, it times out and cannot return results.  This is because the debug stuff is very low priority task.  It really was for debugging when developing .. this is why it is not documented.  You can't really use it on a "live" server that has clients or that is synching, etc.

**Comment by thevops** [comment]
Ok, anyway thanks for your answer.

**Comment by cculianu** [comment]
Is your node busy synching? If it's synching the debug stuff never works because node is just too busy with higher priority work. If it's just serving requests, some of the `/debug?x=y` stuff works sometimes if you get lucky.

But while synching.. never. :)

**Comment by thevops** [comment]
Nope, my node is already synced :)
I've configured metrics exporter (https://github.com/EchterAgo/fulcrum-prometheus-exporter), so debug endpoint won't be necessary. Metrics should be enough for checking if it's working.

**Comment by cculianu** [comment]
Ah.. I see. Yeah debug is useless anyway unless you are ... developing the node and want to verify the state it maintains is sane.
